domain/application/state/application.go

```diff
@@ -397,131 +406,144 @@ func (st *ApplicationState) GetUnitUUID(ctx domain.AtomicContext, unitName strin
 	}
 
 	createParams.Name = args.UnitName
 
-	if err := tx.Query(ctx, createNodeStmt, createParams).Run(); err != nil {
-		return errors.Annotatef(err, "creating net node row for unit %q", args.UnitName)
-	}
-	if err := tx.Query(ctx, createUnitStmt, createParams).Run(); err != nil {
-		return errors.Annotatef(err, "creating unit row for unit %q", args.UnitName)
-	}
-	if args.CloudContainer != nil {
-		if err := st.upsertUnitCloudContainer(ctx, tx, args.UnitName, nodeUUID.String(), args.CloudContainer); err != nil {
-			return errors.Annotatef(err, "creating cloud container row for unit %q", args.UnitName)
-		}
-	}
-	return nil
-}
-
-// UpdateUnit updates the specified application unit, returning an error
-// satisfying [applicationerrors.UnitNotFoundError] if the unit doesn't exist.
-func (st *ApplicationState) UpdateUnit(
-	ctx domain.AtomicContext, appID coreapplication.ID, args application.UpsertUnitArg,
-) error {
-	err := domain.Run(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		toUpdate, err := st.getUnit(ctx, tx, args.UnitName)
-		if err != nil {
-			return errors.Trace(err)
-		}
-
-		if args.PasswordHash != nil {
-			toUpdate.PasswordHash = *args.PasswordHash
-			toUpdate.PasswordHashAlgorithmID = 0 //currently we only use sha256
-		}
-
-		updateUnit := `
-UPDATE unit SET
-    life_id = $unitDetails.life_id,
-    password_hash = $unitDetails.password_hash,
-    password_hash_algorithm_id = $unitDetails.password_hash_algorithm_id
-WHERE uuid = $unitDetails.uuid
-`
-		updateUnitStmt, err := st.Prepare(updateUnit, *toUpdate)
-		if err != nil {
-			return errors.Trace(err)
+	err = domain.Run(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		if err := tx.Query(ctx, createNodeStmt, createParams).Run(); err != nil {
+			return errors.Annotatef(err, "creating net node row for unit %q", args.UnitName)
 		}
-
-		if err := tx.Query(ctx, updateUnitStmt, toUpdate).Run(); err != nil {
-			return errors.Annotatef(err, "updating unit row for unit %q", toUpdate.Name)
+		if err := tx.Query(ctx, createUnitStmt, createParams).Run(); err != nil {
+			return errors.Annotatef(err, "creating unit row for unit %q", args.UnitName)
 		}
 		if args.CloudContainer != nil {
-			if err := st.upsertUnitCloudContainer(ctx, tx, toUpdate.Name, toUpdate.NetNodeID, args.CloudContainer); err != nil {
-				return errors.Annotatef(err, "updating cloud container row for unit %q", toUpdate.Name)
+			if err := st.upsertUnitCloudContainer(ctx, tx, args.UnitName, nodeUUID.String(), args.CloudContainer); err != nil {
+				return errors.Annotatef(err, "creating cloud container row for unit %q", args.UnitName)
 			}
 		}
 		return nil
 	})
-	return errors.Annotatef(err, "updating unit %q for application %q", args.UnitName, appID)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	if err := st.SaveUnitAgentStatus(ctx, unitUUID.String(), args.AgentStatus); err != nil {
+		return "", errors.Annotatef(err, "saving agent status for unit %q", args.UnitName)
```

Comment:
Is it better to use the unit id here as that is the "entity" being operated on not the name ?

Reply:
Users would prefer to see a name in an error string